pcapgo/capture.go

```diff
@@ -0,0 +1,48 @@
+package pcapgo
+
+import (
+	"net"
+	"syscall"
+	"time"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/mdlayher/raw"
+)
+
+// EthernetHandle wraps a raw.Conn, implementing gopacket.PacketDataSource so
+// that the handle can be used with gopacket.NewPacketSource.
+type EthernetHandle struct {
+	*raw.Conn
+}
+
+// ReadPacketData implements gopacket.PacketDataSource.
+func (h *EthernetHandle) ReadPacketData() ([]byte, gopacket.CaptureInfo, error) {
+	b := make([]byte, 4096) // TODO(correctness): how much space do we need?
+	n, _, err := h.ReadFrom(b)
+	if err != nil {
+		return nil, gopacket.CaptureInfo{}, err
+	}
+	packet := gopacket.NewPacket(b[:n], layers.LayerTypeEthernet, gopacket.Default)
+	data := packet.Data()
+	return data, gopacket.CaptureInfo{
+		CaptureLength: len(data),
+		Length:        len(data),
+		Timestamp:     time.Now(),
+	}, nil
+}
+
+// OpenEthernet implements pcap.OpenLive for ethernet interfaces only.
+func OpenEthernet(ifname string) (*EthernetHandle, error) {
```

Comment:
to make things perfectly clear, maybe "NewEthernetHandle(...)"

Reply:
Done.